src/access/AccessControl.h

```diff
@@ -63,28 +63,36 @@ class AccessControl
             virtual void Release() {}
 
             // Simple getters
-            virtual CHIP_ERROR GetAuthMode(AuthMode & authMode) const { return CHIP_ERROR_NOT_IMPLEMENTED; }
-            virtual CHIP_ERROR GetFabricIndex(FabricIndex & fabricIndex) const { return CHIP_ERROR_NOT_IMPLEMENTED; }
-            virtual CHIP_ERROR GetPrivilege(Privilege & privilege) const { return CHIP_ERROR_NOT_IMPLEMENTED; }
+            virtual CHIP_ERROR GetAuthMode(AuthMode & authMode) const { return CHIP_NO_ERROR; }
```

Comment:
But the problem here is that now we are returning no error and not initializing the outparam, so the consumer will work with random memory?

What are the consumers that are OK with this behavior?

Reply:
No real code calls this without a real delegate configured. Actually very few places use access control, by design.

Access Control check is called regardless of whether a delegate is configured (it's baked into the system, not optional) but as you can see from your other comment, it does not check entries, auth modes, etc. and therefore can't accidentally elevate an auth mode or crash because of a bad value.

So IM is a consumer of check only. The only other code doing things with access control (creating entries etc.) is:

- access control cluster, which performs all "management" of the module, but also is responsible for its startup and shutdown
- AddNOC during commissioning, to create a single entry for a new fabric
  - in production, or any tests for both commissioning and access control, this will create an actual entry
  - in any other tests, this will silently fail to create the entry, like a mock would
- access control tests, which configure a delegate for testing (either the production one, or possibly another one)

I can go through and initialize more of these variables, at maybe slight cost of code increase size.

Reply:
I guess my question is whether just making Check() return success is enough or whether we need these changes too.    What are the tests that fail if Check() returns success but this code here keeps returning errors?